token/metadata.go

```diff
@@ -48,3 +53,178 @@ func (m *Metadata) SpentTokenID() []*token2.ID {
 	}
 	return res
 }
+
+// FilterBy returns a new Metadata containing only the metadata that matches the given enrollment IDs.
+// For Issue actions: the returned metadata will contain only the outputs whose owner has the given enrollment IDs.
+// For Transfer actions:
+// - The returned metadata will contain only the outputs whose owner has the given enrollment IDs;
+// - The senders are all included;
+// - The list of token IDs will be empty.
+func (m *Metadata) FilterBy(eIDs ...string) (*Metadata, error) {
+	res := &Metadata{
+		tms:                  m.tms,
+		tokenRequestMetadata: &api2.TokenRequestMetadata{},
+	}
+
+	// filter issues
+	for _, issue := range m.tokenRequestMetadata.Issues {
+		// By default, an issue action contain only one receiver
+
```

Comment:
This comment should probably be moved right next if search(eIDs, recipientEID) == -1